src/CattyTests/FileManager/CBFileManagerTests.swift

```diff
@@ -121,4 +121,113 @@ final class CBFileManagerTests: XCTestCase {
         XCTAssertFalse(result)
         XCTAssertFalse(Project.projectExists(withProjectID: projectId))
     }
+
+    func testLoadPreviewImageAndCacheWhenManualScreenshotCached() {
+        let info = ProjectLoadingInfo.init(forProjectWithName: kDefaultProjectBundleName, projectID: kNoProjectIDYetPlaceholder)!
+        let project = Project(loadingInfo: info)!
+
+        let manualScreenshotThumbnailPath = project.projectPath() + kScreenshotThumbnailPrefix + kScreenshotManualFilename
+        let manualScreenshot = UIImage(color: UIColor.green)!
+
+        let automaticScreenshotThumbnailPath = project.projectPath() + kScreenshotThumbnailPrefix + kScreenshotAutoFilename
+        let automaticScreenshot = UIImage(color: UIColor.red)!
+
+        let imageCache = RuntimeImageCacheMock(thumbnails: [:],
+                                               cachedImages: [manualScreenshotThumbnailPath: manualScreenshot, automaticScreenshotThumbnailPath: automaticScreenshot])
+
+        let fileManager = CBFileManagerMock(imageCache: imageCache)
+
+        let expectation = XCTestExpectation(description: "Load image from cache - Manual Screenshot")
+        print("Running testLoadPreviewImageAndCacheWhenManualScreenshotCached")
+
+        _ = fileManager.loadPreviewImageAndCache(projectLoadingInfo: info) { image, path in
+            XCTAssertEqual(manualScreenshotThumbnailPath, path)
+            XCTAssertEqual(manualScreenshot, image)
+
+            expectation.fulfill()
+        }
+
+        wait(for: [expectation], timeout: 1.0)
+    }
+
+    func testLoadPreviewImageAndCacheWhenAutomaticScreenshotCached() {
+        let info = ProjectLoadingInfo.init(forProjectWithName: kDefaultProjectBundleName, projectID: kNoProjectIDYetPlaceholder)!
+        let project = Project(loadingInfo: info)!
+
+        let automaticScreenshotThumbnailPath = project.projectPath() + kScreenshotThumbnailPrefix + kScreenshotAutoFilename
+        let automaticScreenshot = UIImage(color: UIColor.red)!
+
+        let imageCache = RuntimeImageCacheMock(thumbnails: [:],
```

Comment:
Please also add the `automaticScreenshot` to the thumbnails but with a different `UIImage` to verify that `cachedImages` have priority.